examples/providers/examples/ws.rs

```diff
@@ -0,0 +1,32 @@
+//! Example of using the WS provider to subscribe to new blocks.
+
+use alloy_network::Ethereum;
+use alloy_provider::{Provider, RootProvider};
+use alloy_rpc_client::{RpcClient, WsConnect};
+use eyre::Result;
+use futures_util::StreamExt;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let rpc_url = "wss://eth-mainnet.g.alchemy.com/v2/your-api-key";
+
+    let ws_transport = WsConnect::new(rpc_url);
+    // To use auth
+    // let auth = alloy_transport::Authorization::bearer("token");
+    // let ws_transport = WsConnect::with_auth(rpc_url, Some(auth));
```

Comment:
are we missing this feature?
if so we should track on alloy, if it doesn't exists yet

Reply:
we have it i believe https://alloy-rs.github.io/alloy/alloy/transports/ws/struct.WsConnect.html#method.with_auth but not using with this

Reply:
Yeah we do have it. 
I'll setup a different example for it. 
Left it their to remind myself